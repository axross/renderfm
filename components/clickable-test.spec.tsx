import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import {
  ClickableText,
  ClickableTextProps,
  TextAnchor,
} from "~/components/clickable-text";

describe("<ClickableText>", () => {
  function setup(additionalProps?: Partial<ClickableTextProps>) {
    render(
      <ClickableText data-testid="clickable-text" {...additionalProps}>
        Hello
      </ClickableText>
    );

    return {
      clickableText: () => screen.getByTestId("clickable-text"),
    };
  }

  it("renders a focusable inline element", () => {
    const { clickableText } = setup();

    userEvent.tab();

    expect(clickableText()).toHaveFocus();

    userEvent.tab({ shift: true });

    expect(clickableText()).not.toHaveFocus();

    userEvent.click(clickableText());

    expect(clickableText()).toHaveFocus();

    expect(clickableText()).toMatchSnapshot();
  });
});

describe("<TextAnchor>", () => {
  function setup(additionalProps?: Partial<ClickableTextProps>) {
    render(
      <TextAnchor
        href="https://example.kohei.dev/"
        data-testid="text-anchor"
        {...additionalProps}
      >
        Hello
      </TextAnchor>
    );

    return {
      textAnchor: () => screen.getByTestId("text-anchor"),
    };
  }

  it("renders focusable inline element", () => {
    const { textAnchor } = setup();

    userEvent.tab();

    expect(textAnchor()).toHaveFocus();

    userEvent.tab({ shift: true });

    expect(textAnchor()).not.toHaveFocus();

    userEvent.click(textAnchor());

    expect(textAnchor()).toHaveFocus();

    expect(textAnchor()).toMatchSnapshot();
  });
});
