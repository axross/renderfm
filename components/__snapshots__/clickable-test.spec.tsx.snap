// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ClickableText> renders a focusable inline element 1`] = `
<span
  class="outline-none focus:ring-4 focus:ring-yellow-500/50 transition-all rounded-sm text-yellow-600 dark:text-yellow-400 hover:underline cursor-pointer"
  data-testid="clickable-text"
  tabindex="0"
>
  Hello
</span>
`;

exports[`<TextAnchor> renders focusable inline element 1`] = `
<a
  class="outline-none focus:ring-4 focus:ring-yellow-500/50 transition-all rounded-sm text-yellow-600 dark:text-yellow-400 hover:underline"
  data-testid="text-anchor"
  href="https://example.kohei.dev/"
>
  Hello
</a>
`;
